### PR TITLE
update the schema to latest

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
This is the first step of migrating to the latest clean schema of Application Insights. There are lots of things to do next: add new fields, deprecate and remove the use of performance counters, clean up context tags. But this PR is sufficient by itself. Other changes will make it too big

#389

### CODE REVIEW

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/ExceptionData.java [hunk 1]
```diff
@@ -1,175 +1,196 @@
...
+        
         writer.write("exceptions", exceptions);
-        writer.write("measurements", measurements);
-
-        if (severityLevel != null) {
-            writer.write("severityLevel", severityLevel.toString());
-        }
-
+        
+        writer.write("severityLevel", severityLevel);
```
Comment (Dmitry-Matveev) [inline core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/ExceptionData.java]: Does write() check for null or severityLevel cannot be null now?

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/MessageData.java [hunk 2]
```diff
@@ -1,131 +1,150 @@
...
         writer.write("ver", ver);
+        
         writer.write("message", message);
-
-        if (severityLevel != null) {
-            writer.write("severityLevel", severityLevel.toString());
-        }
-
+        
+        writer.write("severityLevel", severityLevel);
```
Comment (Dmitry-Matveev) [inline core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/MessageData.java]: Same here.

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/DataPoint.java [hunk 3]
```diff
@@ -1,179 +1,218 @@
...
      * Serializes the beginning of this object to the passed in writer.
      * @param writer The writer to serialize this object to.
-     * @throws IOException Might be throw during serialization.
      */
-    protected void serializeContent(JsonTelemetryDataSerializer writer) throws IOException {
+    protected void serializeContent(JsonTelemetryDataSerializer writer) throws IOException
+    {
         writer.write("name", name);
-
-        if (!DataPointType.Measurement.equals(kind)) {
```
Comment (SergeyKanzhelev) [inline core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/DataPoint.java]: I think it was an optimization to not send the value that has a default: https://github.com/Microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/Schemas/Bond/DataPoint.bond#L13 Shouldn't break anything, I'll thing how to improve code generator

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/ExceptionData.java (hunk 1 above)
Comment (SergeyKanzhelev) [inline core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/ExceptionData.java]: https://github.com/Microsoft/ApplicationInsights-Java/pull/396/files#diff-dc6856d601a520155e7f485732bcbf45R90

File: core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/MessageData.java (hunk 2 above)
Comment (SergeyKanzhelev) [inline core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/MessageData.java]: same - write will now check it

**Comment by SergeyKanzhelev** [comment]
@Dmitry-Matveev  thank you for reviewing such a long PR!

**Pull request merged**